docs/security/self-assessment.md

```diff
@@ -0,0 +1,167 @@
+
+This document serves to provide KubeEdge users with an initial understanding of KubeEdge's security, where to find existing security documentation, KubeEdge plans for security, and general overview of KubeEdge security practices, both for development of KubeEdge as well as security of KubeEdge.
+
+This document provides the CNCF TAG-Security with an initial understanding of KubeEdge to assist in a joint-review, necessary for projects under incubation.  Taken together, this document and the joint-review serve as a cornerstone for if and when KubeEdge seeks graduation and is preparing for a security audit.
+
+## Security functions and features
+
+* Critical.  The cloud component CloudHub exposes ports to provide external services, accept connections from edge nodes, and be responsible for cloud-side communication. During communication, webSocket or Quick protocol is used for data transmission, and TLS certificate is used for data encryption.
+* Critical.  The EdgeHub component exposes the port to provide services externally, and provides list&Watch capability externally on the edge side.
+
+## Project compliance
+
+KubeEdge does not currently document meeting particular compliance standards.
+
+## Secure development practices
+
+KubeEdge has achieved the passing level criteria for [CII Best Practices](https://bestpractices.coreinfrastructure.org/en/projects/3018).
+
+### Development Pipeline
+
+All code is maintained in [Git](https://github.com/kubeedge/kubeedge) and changes must be reviewed by maintainers and must pass all unit and e2e tests. Code changes are submitted via Pull Requests (PRs) and must be signed. Commits to the `main` branch directly are not allowed.
+
+### Communication Channels
+
+* Internal. [GitHub Discussions](https://github.com/kubeedge/kubeedge/discussions)
+* Inbound. How do users or prospective users communicate with the team?
+
+- Inbound: [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI), [mailing list](https://groups.google.com/forum/#!forum/kubeedge)
+- Outbound: [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI), [mailing list](https://groups.google.com/forum/#!forum/kubeedge)
+- Security email group: [kubeedge-security](mailto:cncf-kubeedge-security@lists.cncf.io), [distributors announce list](mailto:cncf-kubeedge-distrib-announce@lists.cncf.io)
+
+### Ecosystem
+
+KubeEdge is based on Kubernetes base, 100% compatible with Kubernetes, support declarative API, CRD, scalability. Build cloud-side consistent containers and Kubernetes ecology, application portability. Cloud capabilities are extended to the edge through edge cloud collaboration, including AI collaboration, data collaboration, application collaboration and management collaboration.
```

Comment:
Not only integration and compatibility with Kubernetes, but also include how KubeEdge is exploring the frontiers of Cloud Native, and helping more users from different industry to adpot cloud native technologies.